lib/rust/zephyr-sys/build.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2024 Linaro LTD
+// SPDX-License-Identifier: Apache-2.0
+
+// Pre-build code for zephyr module.
+
+// This module makes the values from the generated .config available as conditional compilation.
+// Note that this only applies to the zephyr module, and the user's application will not be able to
+// see these definitions.  To make that work, this will need to be moved into a support crate which
+// can be invoked by the user's build.rs.
+
+// This builds a program that is run on the compilation host before the code is compiled.  It can
+// output configuration settings that affect the compilation.
+
+use anyhow::Result;
+
+use bindgen::Builder;
+
+use std::env;
+use std::path::PathBuf;
+
+fn main() -> Result<()> {
+    // Pass in the target used to build the native code.
+    let target_arg = format!("--target={}", env::var("TARGET")?);
+
+    // println!("includes: {:?}", env::var("INCLUDE_DIRS"));
+    // println!("defines: {:?}", env::var("INCLUDE_DEFINES"));
+
+    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
+    let wrapper_path = PathBuf::from(env::var("WRAPPER_FILE").unwrap());
+
+    // Bindgen everything.
+    let bindings = Builder::default()
+        .header("wrapper.h")
+        .use_core()
+        .clang_arg(&target_arg);
+    let bindings = define_args(bindings, "-I", "INCLUDE_DIRS");
+    let bindings = define_args(bindings, "-D", "INCLUDE_DEFINES");
+    let bindings = bindings
+        .wrap_static_fns(true)
+        .wrap_static_fns_path(wrapper_path)
+        // <inttypes.h> seems to come from somewhere mysterious in Zephyr.  For us, just pull in the
+        // one from the minimal libc.
```

Comment:
The default configuration in Zephyr has `CONFIG_ENFORCE_ZEPHYR_STDINT` active which adds `-imacros${ZEPHYR_BASE}/include/zephyr/toolchain/zephyr_stdint.h`. Do we need to consider this as well?

Reply:
I can look at pulling in this file instead of one from libc.